code/game/machinery/telecomms/computers/telemonitor.dm

```diff
@@ -1,126 +1,156 @@
-/*
-	Telecomms monitor tracks the overall trafficing of a telecommunications network
-	and displays a heirarchy of linked machines.
-*/
-
 #define MAIN_VIEW 0
 #define MACHINE_VIEW 1
 #define MAX_NETWORK_ID_LENGTH 15
 
 /obj/machinery/computer/telecomms/monitor
 	name = "telecommunications monitoring console"
-	icon_screen = "comm_monitor"
 	desc = "Monitors the details of the telecommunications network it's synced with."
 
-	/// Current screen the user is viewing
-	var/screen = MAIN_VIEW
-	/// Weakrefs of the machines located by the computer
-	var/list/machine_list = list()
+	circuit = /obj/item/circuitboard/computer/comm_monitor
+
+	icon_screen = "comm_monitor"
+
 	/// Weakref of the currently selected tcomms machine
-	var/datum/weakref/selected_machine_ref
+	var/datum/weakref/selected_machine_ref = null
+	/// Weakrefs of the machines located by the computer
+	var/list/datum/weakref/machine_refs
+
+	/// Currently displayed "tab"
+	var/screen = MAIN_VIEW
 	/// The network to probe
 	var/network = "NULL"
 	/// Error message to show
-	var/error_message = ""
-	circuit = /obj/item/circuitboard/computer/comm_monitor
+	var/status_message = null
 
-/obj/machinery/computer/telecomms/monitor/ui_data(mob/user)
-	var/list/data = list(
-		"screen" = screen,
-		"network" = network,
-		"error_message" = error_message,
+/obj/machinery/computer/telecomms/monitor/Initialize(mapload, obj/item/circuitboard/C)
+	. = ..()
+	LAZYINITLIST(machine_refs)
+
+/obj/machinery/computer/telecomms/monitor/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/spritesheet/telecomms)
 	)
 
+/obj/machinery/computer/telecomms/monitor/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+
+	ui = SStgui.try_update_ui(user, src, ui)
```

Comment:
```suggestion
	ui = SStgui.try_update_ui(user, src, ui)
```
If parent gets a UI we don't want the computer to open 2 UIs at once, probably

Reply:
This is how computers are working, and I don't want to mess with them.

![image](https://github.com/tgstation/tgstation/assets/137328283/bcb11bd2-2a0d-4bff-8e53-cdf6b14406f0)
![image](https://github.com/tgstation/tgstation/assets/137328283/2f2e22ee-0b8d-4249-9ce1-717d97a9683a)

Reply:
My bad, didn't think base computer did anything